ports/unix/qstrdefsport: Add variant hook.

dlech opened the pull request:
This adds a new `MICROPY_VARIANT_QSTR_DEFS_H` option to allow variants to add additional qstrs to the unix port.

dpgeorge commented:
Maybe it's better to make `QSTR_DEFS` use `+=`, then `mpconfigvariant.mk` can add to this list?  (That's currently possible for stm32.)

dlech committed:
unix/Makefile: Allow variants to add QSTR_DEFS.

This modifies the unix port Makefile to use += for QSTR_DEFS and
QSTR_GLOBAL_DEPENDENCIES so that variants can add additional files if
needed (similar to stm32 port).

Signed-off-by: David Lechner <david@pybricks.com>

dlech committed:
windows/Makefile: Allow variants to add QSTR_DEFS.

This modifies the windows port Makefile to use += for QSTR_DEFS and
QSTR_GLOBAL_DEPENDENCIES so that variants can add additional files if
needed (similar to stm32 port).

Signed-off-by: David Lechner <david@pybricks.com>

dlech commented:
Good idea. I have updated the PR with that change for the unix and windows ports.
